include/boost/geometry/algorithms/detail/disjoint/areal_areal.hpp

```diff
@@ -141,10 +116,11 @@ struct areal_box
                              Box const& box,
                              Strategy const& strategy)
     {
-        if ( ! for_each_segment(geometry::segments_begin(areal),
-                                geometry::segments_end(areal),
-                                box,
-                                strategy.get_disjoint_segment_box_strategy()) )
+        auto ds = strategy.get_disjoint_segment_box_strategy();
+        if (! geometry::all_segments_of(areal, [&](auto const& s)
```

Comment:
Nitpicks:
- could be `auto const ds` as it's passed to `apply` via reference to `const`, AFAIR.
- capture `[&]` could be `[&box, &ds]` for clarity and, theoretically as I have not checeked, optimisation

Reply:
I'll add `const`.

I used `&` to have the line short as possible. Do you think lambdas automatically store references to all local variables?

Reply:
The lazy `[&]` captures everything by reference and the compiler should be clever enough to optimise away anything that is not explicitly referenced inside the lambda.

I aimed to provoke some thoughts on which one is clearer for a reader, but yeah, shorter seems better.

Reply:
I don't have an opinion about it actually. If the compiler captures everything then maybe it'd be a good idea to explicitly limit it and not use the default captures at all. This would be a premature optimization though. This needs more research.

Reply:
I'm sure things are optimised away in this case.
My suggestion was of the same category as declare/include what you use, declare variables in as narrow scope as possible.
Anyway, thanks for comments, it's not an issue that needs any fixing for me.